### PR TITLE
#7593 Document Builder custom styles Slice 3: Attach stylesheets to the rendered document

### LINKED ISSUE #7493: Document Builder custom styles Slice 3: Attach stylesheets to the rendered document
[Slice spec here](https://www.notion.so/pixiebrix/Phase-3-Enable-custom-button-colors-in-the-Document-Builder-12a2a23b2e4a46158b53d0d04036a290?pvs=4#84a55abd2819413ca680dba52e7d6656)

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #7493 

## Demo

https://www.loom.com/share/c9f441bfb11146118920b3816a5328e8

## Future Work
- I spent a while trying to get a document render test working with a nested document and stylesheets applied, but couldn't figure out how to get rid of the "delayed render" effect. I tried fake timers, `await tick()`, `sleep`, and mocking out various hooks and components, but nothing seemed to work, the rendered test always had the content hidden and a loading indicator showing. So, future work is to figure that out and actually get a test working to show that the stylesheet link is rendered in the document view. I verified this manually for now.

## Checklist

- [ ] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer - @grahamlangford

### CODE REVIEW

File: src/bricks/renderers/document.ts [hunk 1]
```diff
@@ -102,6 +103,7 @@ export class DocumentRenderer extends RendererABC {
       Component: DocumentViewLazy,
       props: {
         body,
+        stylesheets: compact(stylesheets).filter((url) => isValidUrl(url)),
```
Comment (twschiller) [inline src/bricks/renderers/document.ts:106]: I think we want to throw BusinessError on any invalid URLs vs. silently ignoring?

File: src/bricks/renderers/documentView/DocumentView.tsx [hunk 2]
```diff
@@ -50,7 +51,12 @@ const DocumentView: React.FC<DocumentViewProps> = ({
           // DocumentView.css is an artifact produced by webpack, see the DocumentView entrypoint included in
           // `webpack.config.mjs`. We build styles needed to render documents separately from the rest of the sidebar
           // in order to isolate the rendered document from the custom Bootstrap theme included in the Sidebar app
-          href={["/DocumentView.css", bootstrap, bootstrapOverrides]}
+          href={[
+            "/DocumentView.css",
+            bootstrap,
+            bootstrapOverrides,
+            ...stylesheets,
```
Comment (twschiller) [inline src/bricks/renderers/documentView/DocumentView.tsx:58]: Open for the definition of done debate, but my original intent was to exclude the base bootstrap stylesheet if any stylesheets are applied

**Review by twschiller — APPROVED** [review]
Approved to unblock dependent slices. See specification clarification here: https://github.com/pixiebrix/pixiebrix-extension/pull/7507#discussion_r1475312699

File: src/bricks/renderers/documentView/DocumentView.tsx (hunk 2 above)
Comment (BLoe) [reply src/bricks/renderers/documentView/DocumentView.tsx:58]: Ok, I didn't see that in the spec, but happy to make that change here. 👍

Comment (BLoe) [reply src/bricks/renderers/documentView/DocumentView.tsx:58]: Actually, there are some implications of this that would change the scope of this quite a bit I think. I'll make a new slice to address this at the end.

**Pull request merged**